app/static/admin/js/pages/analytics/graph/analytics-graph.js

```diff
@@ -0,0 +1,66 @@
+"use strict";
+
+define(["../analytics-util", "../constants"], function(util, c) {
+
+  function Graph(elementId, keyName, valueName, startDate, endDate, color) {
+    this._elementID = elementId;
+    this._keyName = keyName;
+    this._valueName = valueName;
+    this._startDate = startDate;
+    this._endDate = endDate;
+    this._color = color;
+    this._chart = undefined;
+  }
+
+  Graph.prototype.getKeysAndValues = function() {
+    var that = this;
+
+    return _getAppropriateData().then(function (data) {
+      var keys = _getKeys(data);
+      var values = _getValues(data);
+      return {keys: keys, values: values}
+    });
+
+    /*
+    Private Functions
+     */
+
+    function _getAppropriateData() {
```

Comment:
I get why you did this, but I am not sure if it should really be here. It should optimally be with wherever the data is being used. (as in Graph or any XGraph should be agnostic of data and data should be an input)